contracts/Curation.sol

```diff
@@ -0,0 +1,415 @@
+    uint64 public applyStageLen;
+    uint256 public dispensationPct;
+
+    bytes32 constant public CHANGE_PARAMS_ROLE = keccak256("CHANGE_PARAMS_ROLE");
+    bytes32 constant public CHANGE_VOTING_APP_ROLE = keccak256("CHANGE_VOTING_APP_ROLE");
+
+    // TODO!!!
+    enum TimeUnit { Blocks, Seconds }
+
+    struct Application {
+        address applicant;
+        uint64 date;
+        bool registered;
+        bytes data;
+        uint256 amount;
+        uint256 lockId;
+    }
+
+    struct Challenge {
+        address challenger;
+        uint64 date;
+        bool resolved;
+        uint256 amount;
+        uint256 lockId;
+        uint256 voteId;
+        uint256 dispensationPct;
+        mapping(address => bool) claims; // participants who already claimed their reward
+    }
+
+    mapping(bytes32 => Application) applications;
+    mapping(bytes32 => Challenge) challenges;
+    mapping(address => mapping(uint256 => bool)) usedLocks;
+
+    event NewApplication(bytes32 entryId, address applicant);
+    event NewChallenge(bytes32 entryId, address challenger);
+    event ResolvedChallenge(bytes32 entryId, bool result);
+
+    /**
+     * @notice Initializes Curation app with
+     * @param _registry TODO
+     */
+    function initialize(
+        IRegistry _registry,
+        IStaking _staking,
+        IVoting _voting,
+        uint256 _minDeposit,
+        uint64 _applyStageLen,
+        uint256 _dispensationPct
+    )
+        onlyInit
+        external
+    {
+        initialized();
+
+        require(isContract(_registry));
+        require(isContract(_staking));
+
+        registry = _registry;
+        staking = _staking;
+        _setVotingApp(_voting);
+
+        _setMinDeposit(_minDeposit);
+        _setApplyStageLen(_applyStageLen);
+        _setDispensationPct(_dispensationPct);
+    }
+
+    function newApplication(bytes data, uint256 lockId) isInitialized public returns (bytes32 entryId) {
+        entryId = keccak256(data);
+
+        require(data.length != 0);
+        // check data doesn't have an ongoing application
+        require(!applicationExists(entryId));
+        // check data doesn't exist in Registry
+        require(!registry.exists(entryId));
+
+        // check locked tokens
+        uint256 amount = _checkLock(msg.sender, lockId, MAX_UINT64);
+
+        applications[entryId] = Application({
+            applicant: msg.sender,
+            date: getTimestamp(),
+            registered: false,
+            data: data,
+            amount: amount,
+            lockId: lockId
+        });
+
+        // register used lock
+        NewApplication(entryId, msg.sender);
+    }
+
+    function challengeApplication(bytes32 entryId, uint256 lockId) isInitialized public returns(bytes32) {
+        // check application doesn't have an ongoing challenge
+        require(!challengeExists(entryId));
+        // check locked tokens
+        uint256 amount = _checkLock(msg.sender, lockId, getTimestamp().add(applyStageLen));
```

Comment:
This amount is never checked (it is just saved in the Challenge struct) so I assume there is no minimum amount to stake in order to challenge. Couldn't this become a spam issue in which someone can just challenge all applications  without having to stake and risk virtually any money?

Reply:
It's checked inside `_checkLock`.

Reply:
Totally missed it, 👍